From fc04488b427e2e9ee5e7b76836a018b062b5cd12 Mon Sep 17 00:00:00 2001
From: Isaac Davis <isaac.davis@joyent.com>
Date: Mon, 9 Sep 2019 11:25:48 +0000
Subject: [PATCH] MANTA-4535 `manta-adm update` breaks when deploying
 Prometheus if CNS allow_transfer metadata does not exist yet Reviewed by: Tim
 Foster <tim.foster@joyent.com> Approved by: Tim Foster
 <tim.foster@joyent.com>

---
 lib/deploy.js | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/deploy.js b/lib/deploy.js
index 10c6508..8f76870 100644
--- a/lib/deploy.js
+++ b/lib/deploy.js
@@ -742,7 +742,9 @@ Deployer.prototype.addCnsAllowTransfer = function addCnsAllowTransfer(cb)
 	 * Update the cns service with the vm's admin IP, if necessary.
 	 */
 	function updateCnsSvc(ctx, next) {
-		var existingIps = ctx.cnsSvc.metadata.allow_transfer;
+		var allow_transfer = ctx.cnsSvc.metadata.allow_transfer;
+		var existingIps =
+		    allow_transfer === undefined ? [] : allow_transfer;
 		if (existingIps.indexOf(ctx.ip) > -1) {
 			log.info({
 				svcname: self.svcname,
-- 
2.21.0

